test/inductor/test_cpu_repro.py

```diff
@@ -3394,6 +3395,70 @@ def f(a):
         x = torch.rand(4, 5)
         self.common(f, (x,))
 
+    def test_broadcast_scalar_cpp_tile_2d_kernel(self):
+        # Based on detectron2_maskrcnn backbone (conv2d -> max_pool2d)
+        s0 = 12
+        s1 = 21
+
+        data = torch.randn(
+            [1, 256, 8 * s0, 8 * s1],
+        )
+        weight_one = torch.randn([256, 256, 1, 1], requires_grad=True)
+        weight_two = torch.randn((256, 256, 3, 3), requires_grad=True)
+        bias_one = torch.randn([256], requires_grad=True)
+        bias_two = torch.randn([256], requires_grad=True)
+
+        @torch.compile
+        def fn(data, weight_one, weight_two, bias_one, bias_two):
+            conv_result_one = torch.ops.aten.convolution.default(
+                data,
+                weight_one,
+                bias_one,
+                [1, 1],
+                [1, 1],
+                [1, 1],
+                False,
+                [0, 0],
+                1,
+            )
+
+            conv_result_two = torch.ops.aten.convolution.default(
+                data,
+                weight_two,
+                bias_two,
+                [1, 1],
+                [1, 1],
+                [1, 1],
+                False,
+                [0, 0],
+                1,
+            )
+
+            max_pool_result = torch.nn.functional.max_pool2d(
+                conv_result_one,
+                [1, 1],
+                [2, 2],
+                [0, 0],
+                [1, 1],
+                False,
+            )
+            return conv_result_one, conv_result_two, max_pool_result
+
+        torch._dynamo.mark_dynamic(data, 2)
+        torch._dynamo.mark_dynamic(data, 3)
+        _, (fw_code, _bw_code) = run_and_get_code(
+            fn, data, weight_one, weight_two, bias_one, bias_two
+        )
+
+        kernel_name = "cpp_fused_convolution_max_pool2d_with_indices"
+        self.assertTrue(kernel_name in fw_code)
+
+        conv_maxpool2d_code = fw_code[fw_code.find(kernel_name) :]
+        self.assertTrue("auto tmp1 = static_cast<int8_t>(0);" in conv_maxpool2d_code)
+        self.assertTrue(
+            "auto tmp2 = at::vec::Vectorized<int8_t>(tmp1);" in conv_maxpool2d_code
+        )
```

Comment:
Since it is a compiler failure that you fixed, you may simply use `self.common` to run and check the test. Using string comparison is not very general and we usually want to avoid it as possible.